Media browser rewrite

mpetrovic opened the pull request:
From #7024.

RoySegall commented:
Two things: 
1. The tests are failing because travis could not download them. I re trigger the failing tests.
2. I run the scripts and the bower library downloaded out side openscholar folder:
![selection_383](https://cloud.githubusercontent.com/assets/1222368/6920983/9837ce1a-d7ca-11e4-8036-70d25c925896.png)

I remember i had that one from my PR to get bower. I'll try to fix it.

mpetrovic commented:
Does it put the components relative to the current working directory? I tried using .bowerrc to put them somewhere else.

RoySegall commented:
If the destination library of bower won't be under openscholar then the assets won't be accessible. I'll copy the work from the other branch.

RoySegall commented:
OK. I manged to get the download of the components to the right place.

RoySegall commented on openscholar/modules/frontend/media_browser/media_browser.module:
```diff
@@ -0,0 +1,240 @@
+}
+
+/**
+ * Deprecated
```
What about this one? Don't look right to me.

mpetrovic replied:
Deprecated. I don't like to delete code.

mpetrovic commented:
@RoySegall 
A lot of the tests with the restful stuff are failing. I think this is going to interfere with my work now that I'm focusing on tests for the angular apps.

RoySegall commented:
Something wrong - there is no correlation between the travis results and what we see in the matrix results:
![selection_403](https://cloud.githubusercontent.com/assets/1222368/7511999/3239721c-f4b2-11e4-8f4e-24869afc3d36.png)
![selection_404](https://cloud.githubusercontent.com/assets/1222368/7512002/342007c6-f4b2-11e4-8db8-bcb76c4d4321.png)

I'll merge the last code from api-6811 and see what will happens

RoySegall commented:
@ran81 Can you give this one a look as well?

RoySegall commented:
The tests are failing. maybe some problems with selenium or maybe ajax that didn't complete.

RoySegall commented:
@ran81 Let's try to use selenium as the file upload tool.

ran81 commented:
Update:

This is proving more difficult then seems at first. I found many people asking about how to use selenium to upload files via a system file browser dialog but the solutions I've found were either for a different language such as Java, or didn't work for me.

RoySegall commented:
The question here is what we want to do. Do we still want to try and test the upload of the file or go and check the other functionality of the modal?

rbran100 commented:
Lets not spend any more time on it for now and test the other functionality.

RoySegall commented:
That's it! i managed to get the modal window opened when i click on the button. Now it's using bower to get js components and not put the bower_component library in the root folder. But(!) when clicking on the replace button i get JS error.

mpetrovic commented:
@RoySegall 
I built this on angular-file-upload version 3.2.4. You've updated that to 4.2.4, which uses a different name for the angular service and all the directives. The directives need to be updated before the file uploads will work again.

RoySegall commented:
I'll downgrade and see what will happens.

RoySegall commented:
OK. Now everything works for me except for replacing a file. It's look that in the patch function we passing an empty data:
![selection_432](https://cloud.githubusercontent.com/assets/1222368/7983473/44d7fca4-0ac9-11e5-8908-2ac8f08d503b.png)

RoySegall commented:
@mpetrovic

RoySegall commented on openscholar/modules/os/modules/os_restful/plugins/restful/files/1.0/OsFilesResource.class.php:
```diff
@@ -109,4 +503,156 @@ public function processTermsField($terms) {
+    return parent::putEntity($entity_id);
+  }
+
+  protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missing_fields = FALSE) {
```
Why are we overwriting it?

mpetrovic replied:
Because that's the HTTP method we use when we're replacing a file from the
edit form.

Because using FILE_EXISTS_RENAME in any context makes changes to the
filename without alerting the user, something we've had complaints about
for years.

rbran100 commented on openscholar/modules/os/modules/os_restful/plugins/restful/files/1.0/OsFilesResource.class.php:
```diff
@@ -109,4 +503,156 @@ public function processTermsField($terms) {
+      elseif ($info['property']) {
+        $property_name = $info['property'];
+
+        if (!isset($request[$public_field_name])) {
```
@RoySegall was saying that we should call the (propertyValuesPreprocess) before here so that the field can be altered.

mpetrovic replied:
Altered by what?

RoySegall commented:
Well, i found one problem. It seems that the way we are building the terms is in-consistence:
![selection_445](https://cloud.githubusercontent.com/assets/1222368/8109778/589be974-1061-11e5-9794-6e882c7a711b.png)
Then the method setTerms can't handle the terms preprocess.

RoySegall commented:
@mpetrovic The vocabs fields are missing, any idea?

mpetrovic commented:
@RoySegall
Did any of the AJAX calls return errors?

RoySegall commented:
No, didn't got any errors.

mpetrovic commented:
@RoySegall

Derp. Last week, I changed what EntityService.fetch() returns, and updated
everything but where it's used in the Taxonomy Widget. So the widget is
expecting a different kind of object in it's callback then it's getting.
I'll fix it tomorrow.

On Sun, Jun 14, 2015 at 11:58 AM, Roy Segall <notifications@github.com>
wrote:

> No, didn't got any errors.
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_openscholar_openscholar_pull_7028-23issuecomment-2D111842971&d=BQMCaQ&c=WO-RGvefibhHBZq3fL85hQ&r=bU_C8GMV658Lrb3orcTbrX2-xhEQSIgn0Fxx8FilCF0&m=a9Vgp9GulTXHDt3_SOGt26BzC2WfCcD2UhqSfiQHyFE&s=IKjIq2BIsS9auaI3cXNBLZVqog-f4kbBJFKsbbiwopI&e=>
> .
>

RoySegall commented:
I'm looking on it again and i'm getting the next  error in the file modal after uploading:
![selection_449](https://cloud.githubusercontent.com/assets/1222368/8183588/c8d14bb0-1441-11e5-9bfc-1e48de7418e9.png)
